src/main/java/com/artozersky/HackerNewsAPI/cache/SimpleCache.java

```diff
@@ -0,0 +1,55 @@
+package com.artozersky.HackerNewsAPI.cache;
+
+import com.artozersky.HackerNewsAPI.model.NewsPostModel;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.List;
+import java.util.ArrayList;
+
+public class SimpleCache {
+
+    private final int maxSize;
+    private final Map<Long, NewsPostModel> cache;
+
+    public SimpleCache(int maxSize) {
+        this.maxSize = maxSize;
+        this.cache = new LinkedHashMap<Long, NewsPostModel>(maxSize, 0.75f, true) {
```

Comment:
each time the put method is called this function checks whether the size is exceeded, if indeed exceeded it removes the least accessed entry(removes from the head). because the ctor set to true, it means that the linked list is ordderd by entry access, in the end of the list is the most recently accessed post.